sips/sip-022/sip-022-emergency-pox-fix.md

```diff
@@ -0,0 +1,284 @@
+cycle #58.  There is insufficient time to implement, test, and deploy this
+approach.
+
+# Related Work
+
+Consensus bugs requiring immediate attention such as this
+have been detected and fixed in other blockchains.  In the
+absence of a means of gathering user comments on proposed fixes, the task of
+activating these bugfixes has fallen to miners, exchanges, and node runners.  As
+long as sufficiently many participating entities upgrade, then a chain split is
+avoided and the fixed blockchain survives.  A prominent example was Bitcoin
+[CVE-2010-5139](https://www.cvedetails.com/cve/CVE-2010-5139/), in which a
+specially-crafted Bitcoin transaction could mint arbitrarily many BTC well above
+the 21 million cap.  The [developer
+response](https://bitcointalk.org/index.php?topic=823.0) was to quickly release
+a patched version of Bitcoin and rally enough miners and users to upgrade.  In a
+matter of hours, the canonical Bitcoin chain ceased to include any transactions
+that minted too much BTC.
+
+# Backwards Compatibility
+
+There are no changes to the chainstate database schemas in this SIP.  Everyone
+who runs a Stacks 2.1 node today will be able to run a Stacks 2.2 node off of
+their existing chainstates at the start of reward cycle #58,
+as well as a Stacks 2.3 node off of a Stacks 2.2 chainstate at the start of
+reward cycle #59.
+
+Stacks 2.2 nodes will not interact with Stacks 2.1 nodes on the peer network
+after the Bitcoin block activation height passes.  In addition, Stacks 2.2 nodes
+will ignore block-commits from Stacks 2.1 nodes.  Similar changes were made for
+Stacks 2.05 and Stacks 2.1 to ensure that the new network cleanly separates from
+stragglers still following the old rules.  This also applies to Stacks 2.3 nodes
+and Stacks 2.2 nodes.
+
+# Activation
+
+This SIP shall be considered Activated if the Stacks 2.3 network is live at the
+start of reward cycle #59.
+
+The Bitcoin block activation height for the first hard fork will need to pass prior to the selection of
+the PoX anchor block for reward cycle #58 (Bitcoin block height 787851).  This
+SIP proposes Bitcoin block height 787651, which is 200 Bitcoin blocks prior.  In
+other words, the Bitcoin block that is mined 100 blocks prior to the start of
+the prepare phase for reward cycle #58 shall be the activation height.
+
+The second hard fork will take effect 200 Bitcoin blocks prior to the start of reward cycle #59,
+which is Bitcoin block height 789751.
+
+The node software for Stacks 2.2 and 2.3 shall be merged to the `master` branch of the
+reference implementation no later than three days prior to the activation
+height.  This means that everyone shall have at least three days to upgrade
+their Stacks 2.1 nodes to Stacks 2.2, and at least three days to upgrade from
+2.2 to 2.3.
+
+# Reference Implementation
+
+The reference implementation of this SIP can be found in the
+`feat/address-pox-increase` branch of
```

Comment:
branch for first hard fork is `feat/sip-022-pox-disable` (
https://github.com/stacks-network/stacks-blockchain/pull/3677)

No branch for second hard fork yet.